Projects/b_u585i_iot02a_ntz/Src/ota_pal/ota_pal_stm32u5_ntz.c

```diff
@@ -375,390 +591,565 @@ static HAL_StatusTypeDef prvWriteToFlash( uint32_t destination,
+                            ( prvEraseBank( pxContext->ulTargetBank ) == pdTRUE ) &&
+                            ( prvDeleteImageNvContext() == pdTRUE ) )
+                        {
+                            xOtaStatus = OtaPalSuccess;
+                            pxContext->xPalState = OTA_PAL_REJECTED;
+                        }
+
+                        break;
+
+                    case OTA_PAL_REJECTED:
+                        xOtaStatus = OtaPalSuccess;
+                        break;
+
+                    default:
+                        break;
                 }
 
                 break;
 
-            case OtaImageStateRejected:
+            case OtaImageStateTesting:
 
-                /* Make sure the image is not committed already. A committed image cannot be aborted or rejected. */
-                if( pContext->state != OTA_PAL_IMAGE_STATE_COMMITTED )
-                {
-                    /*
-                     *  Remove the persisted state of image from flash.
-                     */
-                    prvRemoveImageContextFromFlash();
-                    pContext->state = OTA_PAL_IMAGE_STATE_INVALID;
-                    otaStatus = OtaPalSuccess;
-                }
-                else
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalRejectFailed;
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_PENDING_SELF_TEST:
+
+                        /* Select target bank to start testing */
+                        if( prvSelectBank( pxContext->ulTargetBank ) == pdTRUE )
+                        {
+                            xOtaStatus = OtaPalSuccess;
+                            pxContext->xPalState = OTA_PAL_PENDING_SELF_TEST;
+                            ( void ) prvWritePalNvContext( pxContext );
+                        }
+
+                        break;
+
+                    default:
+                        break;
                 }
 
                 break;
 
             case OtaImageStateAborted:
 
-                /* Make sure the image is not committed already. A committed image cannot be aborted or rejected. */
-                if( pContext->state != OTA_PAL_IMAGE_STATE_COMMITTED )
-                {
-                    /*
-                     *  Remove the persisted state of image from flash.
-                     */
-                    prvRemoveImageContextFromFlash();
-                    pContext->state = OTA_PAL_IMAGE_STATE_INVALID;
-                    otaStatus = OtaPalSuccess;
-                }
-                else
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalAbortFailed;
+                    case OTA_PAL_READY:
+                    case OTA_PAL_REJECTED:
+                    case OTA_PAL_ACCEPTED:
+                        xOtaStatus = OtaPalSuccess;
```

Comment:
I think we should return failure if trying to abort an already committed image.

Reply:
From the docs: "This function may be called before the file is opened", thus even if an image is in the "Accepted" state, the ota agent could still call the abort function.